Priority Queue/FastPriorityQueue.cs

```diff
@@ -365,7 +407,8 @@ public void Remove(T node)
             #endif
 
             //If the node is already the last node, we can remove it immediately
-            if(node.QueueIndex == _numNodes)
+            int index = node.QueueIndex;
```

Comment:
Local variables are not intrinsically faster than simple getters/setters _(which can be optimized away)_, except that it may be easier for a compiler to tell that it can optimize a local-variable into a register.

In this case it's able to do it for both _(on the x86 JIT, at least)_, so there's no difference in the output.